src/VecSim/spaces/computer/preprocessor.h

```diff
@@ -14,21 +14,14 @@ class PreprocessorAbstract : public VecsimBaseObject {
 public:
     PreprocessorAbstract(std::shared_ptr<VecSimAllocator> allocator)
         : VecsimBaseObject(allocator) {}
-    struct PreprocessParams;
-    virtual void preprocess(const void *original_blob, MemoryUtils::unique_blob &storage_blob,
-                            MemoryUtils::unique_blob &query_blob,
-                            PreprocessParams &params) const = 0;
-    virtual void preprocessForStorage(MemoryUtils::unique_blob &blob) const = 0;
-    virtual void preprocessQuery(MemoryUtils::unique_blob &blob) const = 0;
-
-    virtual bool hasQueryPreprocessor() const = 0;
-    virtual bool hasStoragePreprocessor() const = 0;
-
-    struct PreprocessParams {
-        const size_t processed_bytes_count;
-        bool is_populated_storage;
-        bool is_populated_query;
-    };
+    virtual void preprocess(const void *original_blob, void *&storage_blob, void *&query_blob,
+                            size_t processed_bytes_count, unsigned char alignment) const = 0;
```

Comment:
shouldn't the `processed_bytes_count` be a reference in case it is changed by the preprocessor?

Reply:
i havn't yet decicded how we would handle a preprocessor that changes the allocation size 
for now we assume uniform size